packages/ui-react/src/components/Sidebar/Sidebar.tsx

```diff
@@ -1,81 +1,36 @@
-import React, { Fragment, useEffect, useRef, useState, type ReactNode } from 'react';
+import React, { Fragment, type ReactNode, type RefObject, useEffect, useRef } from 'react';
 import classNames from 'classnames';
 import { useTranslation } from 'react-i18next';
 import Close from '@jwp/ott-theme/assets/icons/close.svg?react';
 
 import IconButton from '../IconButton/IconButton';
 import Icon from '../Icon/Icon';
-import scrollbarSize from '../../utils/dom';
+import { useModal } from '../../containers/ModalProvider/useModal';
 
 import styles from './Sidebar.module.scss';
 
 type SidebarProps = {
   isOpen: boolean;
   onClose: () => void;
   children?: ReactNode;
+  containerRef?: RefObject<HTMLElement>;
 };
 
-const Sidebar: React.FC<SidebarProps> = ({ isOpen, onClose, children }) => {
+const Sidebar: React.FC<SidebarProps> = ({ isOpen, onClose, containerRef, children }) => {
   const { t } = useTranslation('menu');
-  const lastFocusedElementRef = useRef<HTMLElement | null>(null);
   const sidebarRef = useRef<HTMLDivElement>(null);
-  const [visible, setVisible] = useState(false);
+
+  useModal({ open: isOpen, onClose, modalRef: sidebarRef, containerRef });
 
   const htmlAttributes = { inert: !isOpen ? '' : undefined }; // inert is not yet officially supported in react. see: https://github.com/facebook/react/pull/24730
 
   useEffect(() => {
-    if (isOpen) {
-      // Before inert on the body is applied in Layout, we need to set this ref
-      lastFocusedElementRef.current = document.activeElement as HTMLElement;
-
-      // When opened, adjust the margin-right to accommodate for the scrollbar width to prevent UI shifts in background
-      document.body.style.marginRight = `${scrollbarSize()}px`;
-      document.body.style.overflowY = 'hidden';
-
-      // Scroll the sidebar to the top if the user has previously scrolled down in the sidebar
-      if (sidebarRef.current) {
-        sidebarRef.current.scrollTop = 0;
-      }
-    } else {
-      document.body.style.removeProperty('margin-right');
-      document.body.style.removeProperty('overflow-y');
+    // Scroll the sidebar to the top if the user has previously scrolled down in the sidebar
+    if (isOpen && sidebarRef.current) {
+      sidebarRef.current.scrollTop = 0;
     }
```

Comment:
Don't we want to move this to Modal?

Reply:
Not sure, it's sidebar specific logic and I want to prevent to make changes to existing behaviours